### ISSUE TITLE
feat(DomPortalOutlet): Alternative constructor (ComponentFactoryResolver is deprecated)

### ISSUE THREAD

**Issue by bbarry** [angular/components#24334]
### Feature Description

@angular/angular deprecated `ComponentFactoryResolver` in 13.2.0:

https://github.com/angular/angular/blob/master/CHANGELOG.md#1320-2022-01-26

When using a lint rule such as deprecation/deprecation users of `DomPortalOutlet` now get a warning due to the constructor requiring a `ComponentFactoryResolver`

### Use Case

https://github.com/angular/components/blob/aea5d136068798e8655a9fecaaabd6cb73cb8ab5/src/cdk/portal/portal.spec.ts#L423-L438

I am not sure if the tslint `deprecation` rule is enabled in this repository (it is not in tslint.json here) but I presume it would warn on the above tests. 

It would be nice to not require an 

```typescript
    // eslint-disable-next-line deprecation/deprecation
    private componentFactoryResolver: ComponentFactoryResolver
```

in my code, and presumably the Angular team is planning on removing this at some point.

**Comment by stefnedelchev** [comment]
I wasn't sure if I have to create a new issue so I decided to write here as my issue is kinda related. Although the old constructor is deprecated, I still find it impossible to create a `DomPortalOutlet` by passing `outletElement` only.
![image](https://user-images.githubusercontent.com/88819267/156330855-cf33fde6-5bcb-4a15-a4fb-ded3d989f690.png)
I opened the class definition and I clearly see that it has only one constructor and it's the one with the deprecated params. The version I use is the latest (by now) 13.2.4
